src/main/java/pl/marcin/project/ShopAppRunner.java

```diff
@@ -0,0 +1,10 @@
+package pl.marcin.project;
+
+import pl.marcin.project.controller.Shop;
+
+public class ShopAppRunner implements AppRunner {
+    @Override
+    public void runApplication() {
+        Shop.main(new String[]{});
```

Comment:
instead of calling the `main` method here, we can extract everything that currently is in `Shop.main` into a method and run that newly extracted method here instead of `main`. This way `main` remains untouched :)

Reply:
done